### PR TITLE
Independent toolbar buttons twig component

### LINKED ISSUE #33604: Independent toolbar buttons twig component
A bit more details in #33182

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Try to add independent behaviour on toolbar twig component.<br/>We can't add title to this because of the order of procesesing the BO render.<br/>When we drop the smarty way, this could be improved this component.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI 🟢 & Tests 🟢
| Fixed issue or discussion?     | Fixes #33604
| Related PRs       | 
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Review by mflasquin — DISMISSED** [review]

File: src/PrestaShopBundle/Twig/Component/Toolbar.php [hunk 1]
```diff
@@ -69,4 +71,22 @@ public function mount(): void
...
+    {
+        // todo: Find a better way when we definitively drop smarty...
+        return $this->context->getLegacyLayoutData()->enableSidebar;
+    }
+
+    public function getHelpLink()
+    {
+        // todo: Find a better way when we definitively drop smarty...
+        return $this->context->getLegacyLayoutData()->helpLink;
+    }
```
Comment (M0rgan01) [inline src/PrestaShopBundle/Twig/Component/Toolbar.php:88]: I think we can remove the class variables `$toolbarBtn` `$enableSidebar` `$helpLink`

**Review by jolelievre — CHANGES_REQUESTED** [review]
To discuss tomorrow but I'm not sure we should introduce this intermediate DTO accessible via the legacy context just so that the Twig component can access them differently via a legacy context service as well

Seems more like we are in the situation we discussed previously with variables injected by the controller that can't be fetched automatically by the components, so we should wait for the global layout to handle them and pass them down to the components

**Comment by M0rgan01** [comment]
tests: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/6037635048

**Comment by boherm** [comment]
@jolelievre I agree with you about it :)
So we must close this PR for now.

**Pull request closed without merging**